### PR TITLE
Bug fixes, PEP8 fixes, Python3 compatiblity improvements.

### PR DESCRIPTION
**Pull request by Xazax-hun**
Bug fixes, PEP8 fixes, Python3 compatiblity improvements.

### CODE REVIEW

File: build_package.py [hunk 1]
```diff
@@ -524,15 +529,14 @@ def build_package(repository_root, build_package_config, env=None):
...
-    version_json_data = json.loads(version_data)
+        version_json_data = json.load(v_file)
 
     git_hash = ''
     try:
         git_hash_cmd = ['git', 'rev-parse', 'HEAD']
         git_hash = subprocess.check_output(git_hash_cmd,
                                            cwd=repository_root)
-        git_hash = git_hash.rstrip()
+        git_hash = str(git_hash.rstrip())
```
Comment (bruntib) [inline build_package.py]: Isn't `rstrip()` returning a string anyway?

File: build_package.py [hunk 2]
```diff
@@ -163,17 +167,18 @@ def copy_tree(src, dst):
             copy_tree(source, destination)
         else:
             if not os.path.exists(destination) or \
-                    os.stat(src).st_mtime - os.stat(dst).st_mtime > 1:
+                                    os.stat(src).st_mtime - os.stat(
+                                dst).st_mtime > 1:
```
Comment (bruntib) [inline build_package.py]: Are we sure about this indentation? (Not a major issue, just looks weirder than the original one:) )

File: codechecker_lib/host_check.py [hunk 3]
```diff
@@ -128,10 +125,9 @@ def check_intercept(env):
             return False
 
     except OSError as oerr:
-        if oerr[0] == errno.ENOENT:
-            # not just intercept-build can be used for logging
-            # it is possible that another build logger is available
+        if oerr.errno == errno.ENOENT:
```
Comment (gyorb) [inline codechecker_lib/host_check.py]: Better this way, thanks!

File: codechecker_lib/pgpass.py [hunk 4]
```diff
@@ -3,88 +3,94 @@
...
-      if pw:
-        return pw
+    Returns None on no matching entry, otherwise it returns the password.
 
-  return None
+    For file format see:
+    http://www.postgresql.org/docs/current/static/libpq-pgpass.html
+    """
+
+    if len(hostname) == 0 or len(port) == 0 or len(database) == 0 or len(
```
Comment (gyorb) [inline codechecker_lib/pgpass.py]: Weird split.

File: codechecker_lib/suppress_file_handler.py [hunk 5]
```diff
@@ -69,43 +69,45 @@ def get_suppress_data(suppress_file):
...
     try:
         with codecs.open(suppress_file, 'r', 'UTF-8') as s_file:
             suppress_data = get_suppress_data(s_file)
 
         if not os.stat(suppress_file)[6] == 0:
-            # file is not empty
+            # File is not empty.
 
-            res = filter(lambda x: (x[0] == value and x[1] == file_name) or (x[0] == value and x[1] == ''), suppress_data)
+            res = filter(lambda x: (x[0] == value and x[1] == file_name) or (
```
Comment (gyorb) [inline codechecker_lib/suppress_file_handler.py]: Weird place for line split.

File: codechecker_lib/suppress_file_handler.py [hunk 6]
```diff
@@ -69,43 +69,45 @@ def get_suppress_data(suppress_file):
...
 
             if res:
-                LOG.debug("Already found in\n %s" % (suppress_file))
+                LOG.debug("Already found in\n %s" % suppress_file)
                 return True
 
         s_file = codecs.open(suppress_file, 'a', 'UTF-8')
 
-        s_file.write(value+COMMENT_SEPARATOR+file_name+COMMENT_SEPARATOR+comment+'\n')
+        s_file.write(
```
Comment (gyorb) [inline codechecker_lib/suppress_file_handler.py]: Weird split.

File: db_model/orm_model.py [hunk 7]
```diff
@@ -192,47 +207,67 @@ class Report(Base):
...
 
-
     # Priority/severity etc...
-    def __init__(self, run_id, bug_id, file_id, checker_message, start_bugpoint, start_bugevent, end_bugevent, checker_id, checker_cat, bug_type, severity, suppressed):
+    def __init__(self, run_id, bug_id, file_id, checker_message, start_bugpoint,
+                 start_bugevent, end_bugevent, checker_id, checker_cat,
+                 bug_type, severity, suppressed):
         self.run_id = run_id
         self.file_id = file_id
         self.bug_id, self.checker_message = bug_id, checker_message
```
Comment (gyorb) [inline db_model/orm_model.py]: Could we move these into separate lines?

File: db_model/orm_model.py [hunk 8]
```diff
@@ -192,47 +207,67 @@ class Report(Base):
...
         self.start_bugpoint = start_bugpoint
         self.start_bugevent = start_bugevent
         self.end_bugevent = end_bugevent
         self.severity = severity
-        self.checker_id, self.checker_cat, self.bug_type = checker_id, checker_cat, bug_type
-        self.suppressed = suppressed
+        self.checker_id, self.checker_cat = checker_id, checker_cat
```
Comment (gyorb) [inline db_model/orm_model.py]: Could we move these into separate lines?

File: storage_server/report_server.py [hunk 9]
```diff
@@ -116,17 +114,21 @@ def __del_buildaction_results(self, build_action_id, run_id):
...
-                        LOG.debug("Stored source file needs to be kept, there is reference to it from another report.")
-                        # report needs to be deleted if there is no reference the
-                        # file cascade delete will remove it
-                        # else manual cleanup is needed
+                    self.__sequence_deleter(BugReportPoint,
+                                            report.start_bugpoint)
+
+                    if self.__del_source_file_for_report(run_id, report.id,
+                                                         report.file_id):
+                        LOG.debug(
```
Comment (gyorb) [inline storage_server/report_server.py]: Weird split.

File: storage_server/report_server.py [hunk 10]
```diff
@@ -340,25 +341,26 @@ def storeReportInfo(self,
...
             # Old suppress format did not contain file name.
             suppressed = self.session.query(SuppressBug) \
-                               .filter(
-                                   and_(SuppressBug.run_id == action.run_id,
-                                        SuppressBug.hash == bug_hash,
-                                        or_(SuppressBug.file_name == source_file_name,
-                                            SuppressBug.file_name == u''))) \
-                               .count() > 0
+                             .filter(
+                and_(SuppressBug.run_id == action.run_id,
```
Comment (gyorb) [inline storage_server/report_server.py]: Why is this better than before? Was there some pylint warning?

File: tests/functional/package_test/__init__.py [hunk 11]
```diff
@@ -256,22 +251,11 @@ def _generate_skip_list_file(skip_list_file):
...
-    check_cmd.append('--skip')
-    check_cmd.append(skip_list_file)
-    check_cmd.append('-n')
-    check_cmd.append(test_project_name)
-    check_cmd.append('-b')
-    check_cmd.append("'"+test_project_build_cmd+"'")
-    check_cmd.append('--analyzers')
-    check_cmd.append('clangsa')
-    check_cmd.append('--quiet-build')
+    check_cmd = ['CodeChecker', 'check', '-w', shared_test_params['workspace'],
```
Comment (gyorb) [inline tests/functional/package_test/__init__.py]: I'm not sure this is more readable than before. I would much prefer if a list is used to move each argument into a new line.
```
check_cmd = ['CodeChecker',
             'check',
             '-w', shared_test_params['workspace'],
             ....
```

File: tests/functional/package_test/__init__.py [hunk 12]
```diff
@@ -305,17 +289,11 @@ def start_server_proc(event, server_cmd, checking_env):
...
-    server_cmd.append('server')
-    server_cmd.append('--check-port')
-    server_cmd.append(str(test_config['CC_TEST_SERVER_PORT']))
-    server_cmd.append('--view-port')
-    server_cmd.append(str(test_config['CC_TEST_VIEWER_PORT']))
-    server_cmd.append('-w')
-    server_cmd.append(shared_test_params['workspace'])
-    server_cmd.append('--suppress')
-    server_cmd.append(shared_test_params['suppress_file'])
+    server_cmd = ['CodeChecker', 'server', '--check-port',
```
Comment (gyorb) [inline tests/functional/package_test/__init__.py]: Same here. the corresponding arguments with the values in one line.

File: tests/functional/package_test/test_regression_getall_wrapper.py [hunk 13]
```diff
@@ -47,7 +46,8 @@ def setUp(self):
     def test_get_run_results_no_filter(self):
         """ Get all the run results without any filtering. """
         runid = self._runid
-        logging.debug('Get all run results from the db for runid: ' + str(runid))
+        logging.debug(
```
Comment (gyorb) [inline tests/functional/package_test/test_regression_getall_wrapper.py]: Weird split.

File: tests/functional/package_test/test_regression_getall_wrapper.py [hunk 14]
```diff
@@ -215,7 +215,8 @@ def test_get_run_results_severity_sort(self):
     def test_get_run_results_sorted2(self):
         """ Get the run results and sort them by filename and checkername ASC. """
         runid = self._runid
-        logging.debug('Get all run results from the db for runid: ' + str(runid))
+        logging.debug(
```
Comment (gyorb) [inline tests/functional/package_test/test_regression_getall_wrapper.py]: Weird split. Could be split after the text before the run id.

File: tests/performance_test/run_performance_test.py [hunk 15]
```diff
@@ -6,91 +6,110 @@
```
Comment (gyorb) [inline tests/performance_test/run_performance_test.py]: This file is fully rewritten in the improve performance test pull request.
You could skip the changes here to merge/rebase more easily.

File: viewer_server/client_db_access_server.py [hunk 16]
```diff
@@ -3,22 +3,26 @@
 #   This file is distributed under the University of Illinois Open Source
```
Comment (gyorb) [inline viewer_server/client_db_access_server.py]: This file is modified heavily and cleaned up in the authentication pull request. Might help to skip the modifications now to merge/rebase more easily with the new features.

File: .ci/basic_python_requirements [hunk 17]
```diff
@@ -1,3 +1,3 @@
 sqlalchemy==1.0.9
 alembic==0.8.2
-thrift==0.9.1
+thrift==0.9.3
```
Comment (Xazax-hun) [inline .ci/basic_python_requirements]: 0.9.1 can not be installed with python 3, but in the meantime I realized that python3 generators for thrift will only come for 1.0, so I will just revert this for now.

File: viewer_server/client_db_access_server.py (hunk 16 above)
Comment (Xazax-hun) [inline viewer_server/client_db_access_server.py]: I think these changes will not be too hard to merge.

File: tests/performance_test/run_performance_test.py (hunk 15 above)
Comment (Xazax-hun) [inline tests/performance_test/run_performance_test.py]: Ok.

**Pull request merged**